### PR TITLE
Sugarcrm

### PR DESCRIPTION
**Pull request by jbianquetti-nami**
Initial PR for SugarCRM 

Recommendations that had been made for previous charts are (hopefully) applied to this too.

### CODE REVIEW

File: incubator/sugarcrm/README.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+The command removes all the Kubernetes components associated with the chart and deletes the release.
+
+## Configuration
+
+The following tables lists the configurable parameters of the sugarcrm chart and their default values.
+
+|              Parameter               |               Description                |                         Default                         |
+|--------------------------------------|------------------------------------------|---------------------------------------------------------|
+| `image`                              | sugarcrm image                           | `bitnami/sugarcrm:{VERSION}`                            |
+| `imageTag`                           | sugarcrm image Tag                       | `3.1.3`                                                 |
```
Comment (sameersbn) [inline incubator/sugarcrm/README.md]: we don't need no `imageTag` field, combine with `image` field

File: incubator/sugarcrm/values.yaml [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+##  sugarcrmHost: 
+
+
+
+## SMTP mail delivery configuration
+## ref: https://github.com/bitnami/bitnami-docker-sugarcrm#smtp-configuration
+##
+# smtpHost:
+# smtpPort:
+# smtpHostID:
```
Comment (sameersbn) [inline incubator/sugarcrm/values.yaml]: please remove this comment

File: incubator/sugarcrm/values.yaml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 ## Bitnami sugarcrm image 
 ## ref: https://hub.docker.com/r/bitnami/sugarcrm/tags/
 ##
-image: bitnami/sugarcrm:6.5.24-r2
+image: bitnami/sugarcrm
```
Comment (sameersbn) [inline incubator/sugarcrm/values.yaml]: why?

**Comment by sameersbn** [comment]
FYI, I noticed initialization issues with SugarCRM image and reported the issue here: https://github.com/bitnami/bitnami-docker-sugarcrm/issues/15

File: incubator/sugarcrm/values.yaml [hunk 4]
```diff
@@ -0,0 +1,98 @@
+## Bitnami sugarcrm image 
+## ref: https://hub.docker.com/r/bitnami/sugarcrm/tags/
+##
+image: bitnami/sugarcrm
```
Comment (sameersbn) [inline incubator/sugarcrm/values.yaml]: Please add the image tag here

File: incubator/sugarcrm/values.yaml [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+
+## User of the application
+## ref: https://github.com/bitnami/bitnami-docker-sugarcrm#environment-variables
+##
+sugarcrmLastname: Name
+
+## Application password
+## Defaults to a random 10-character alphanumeric string if not set
+## ref: https://github.com/bitnami/bitnami-docker-sugarcrm#environment-variables
+## sugarcrmPassword:
```
Comment (sameersbn) [inline incubator/sugarcrm/values.yaml]: Please move the description of the `sugarcrmPassword` variable after the `sugarcrmUsername` field.

File: incubator/sugarcrm/values.yaml [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+## ref: http://kubernetes.io/docs/user-guide/images/#pre-pulling-images
+##
+# imagePullPolicy:
+
+## User of the application
+## ref: https://github.com/bitnami/bitnami-docker-sugarcrm#environment-variables
+##
+sugarcrmUsername: user
+
+## User of the application
```
Comment (sameersbn) [inline incubator/sugarcrm/values.yaml]: `User of the application` -> `Last name`. I wonder if the sugarCRM image is missing a first name param.

File: incubator/sugarcrm/values.yaml [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+##
+sugarcrmEmail: user@example.com
+
+## SugarCRM HOST
+## ref: https://github.com/bitnami/bitnami-docker-sugarcrm#environment-variables
+##
+##  sugarcrmHost: 
+
+
+
```
Comment (sameersbn) [inline incubator/sugarcrm/values.yaml]: lets get rid of these extra new line characters.

File: incubator/sugarcrm/values.yaml [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+## Enable persistence using Persistent Volume Claims
+## ref: http://kubernetes.io/docs/user-guide/persistent-volumes/
+##
+persistence:
+  enabled: true
+  sugarcrm:
+    storageClass: generic
+    accessMode: ReadWriteOnce
+    size: 1Gi
```
Comment (sameersbn) [inline incubator/sugarcrm/values.yaml]: please define a volume for apache data

**Comment by sameersbn** [comment]
@jbianquetti-nami it appears that for the sugarcrm application to work correctly, the `sugarcrmHost` should be configured. can you please rebase this chart on the `kubernetes/stable/ghost` chart by @prydonius. It has the same requirements.

File: incubator/sugarcrm/README.md [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+
+The command removes all the Kubernetes components associated with the chart and deletes the release.
+
+## Configuration
+
+The following tables lists the configurable parameters of the sugarcrm chart and their default values.
+
+|              Parameter               |               Description                |                         Default                         |
+|--------------------------------------|------------------------------------------|---------------------------------------------------------|
+| `image`                              | sugarcrm image                           | `bitnami/sugarcrm:{VERSION}`                            |
```
Comment (prydonius) [inline incubator/sugarcrm/README.md]: `sugarcrm image` -> `SugarCRM image`

File: incubator/sugarcrm/Chart.yaml [hunk 10]
```diff
@@ -0,0 +1,14 @@
+name: sugarcrm
+version: 0.1.0
+description: Sugar offers the most innovative, flexible and affordable CRM in the market and delivers the best all-around value of any CRM.
```
Comment (prydonius) [inline incubator/sugarcrm/Chart.yaml]: `SugarCRM` offers?

File: incubator/sugarcrm/README.md [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+
+Specify each parameter using the `--set key=value[,key=value]` argument to `helm install`. For example,
+
+```console
+$ helm install --name my-release \
+  --set sugarcrmUsername=admin,sugarcrmPassword=password,mariadb.mariadbRootPassword=secretpassword \
+    stable/sugarcrm
+```
+
+The above command sets the sugarcrm administrator account username and password to `admin` and `password` respectively. Additionally it sets the MariaDB `root` user password to `secretpassword`.
```
Comment (prydonius) [inline incubator/sugarcrm/README.md]: `sugarcrm` -> `SugarCRM`

File: incubator/sugarcrm/README.md [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+
+```console
+$ helm install --name my-release -f values.yaml stable/sugarcrm
+```
+
+> **Tip**: You can use the default [values.yaml](values.yaml)
+
+## Persistence
+
+The [Bitnami sugarcrm](https://github.com/bitnami/bitnami-docker-sugarcrm) image stores the sugarcrm data and configurations at the `/bitnami/sugarcrm` path of the container.
```
Comment (prydonius) [inline incubator/sugarcrm/README.md]: `sugarcrm` -> `SugarCRM`

File: incubator/sugarcrm/templates/NOTES.txt [hunk 13]
```diff
@@ -0,0 +1,27 @@
+
+1. Get the sugarcrm URL by running:
```
Comment (prydonius) [inline incubator/sugarcrm/templates/NOTES.txt]: Should use the capitalised name

**Comment by sameersbn** [comment]
@jbianquetti-nami fyi, https://github.com/bitnami/bitnami-docker-sugarcrm/issues/16
Abandoning this chart until the issues in the image are fixed.

**Comment by sameersbn** [comment]
added in #310

**Pull request closed without merging**